### PR TITLE
Word count template in Java

### PR DESCRIPTION
**Pull request by jaugustorufino**
Implements a template in Java of the word count program based on [wc_single_proc.c](https://github.com/thiagomanel/SO/blob/master/process/word_count/wc_single_proc.c).

* Execution of `gradle runApp`:
```gradle

> Task :runApp
70400000

BUILD SUCCESSFUL in 9s
2 actionable tasks: 2 executed

```

@thiagomanel

### CODE REVIEW

File: 2023.1/word_count/java/src/main/java/WordCount.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    // │     ├── ...
+    // │     └── file
+    // ├── ...
+    // └── subdir N
+    // │     ├── file
+    // │     ├── ...
+    // │     └── file
+    public static void main(String[] args) {
+        if (args.length != 1) {
+            System.out.println("Usage: java WordCount <root_directory>");
```
Comment (thiagomanel) [inline 2023.1/word_count/java/src/main/java/WordCount.java:32]: i think we can be more explicit here. what do you think?

```System.err.println("Usage: java WordCount <root_directory>");
 System.exit(1);```

**Review by thiagomanel — CHANGES_REQUESTED** [review]
thank @jaugustorufino. I've dropped only one small suggestion on arg handling

File: 2023.1/word_count/java/src/main/java/WordCount.java (hunk 1 above)
Comment (jaugustorufino) [reply 2023.1/word_count/java/src/main/java/WordCount.java:32]: Sure, I agree. Thank you for the feedback!

**Comment by jaugustorufino** [comment]
I just noticed that my IDE is complaining about the missing package declaration in Java Class.
image
But when I put this declaration, I've got this executing `gradle runApp`:
```gradle

> Task :runApp FAILED
Error: Could not find or load main class WordCount
Caused by: java.lang.ClassNotFoundException: WordCount

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':runApp'.
> Process 'command '/usr/lib/jvm/java-11-openjdk-amd64/bin/java'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.5.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 579ms
2 actionable tasks: 2 executed
```
What do you think? @thiagomanel

**Review by thiagomanel — APPROVED** [review]
lgtm

**Pull request merged**